lib/remote_storage/swift.rb

```diff
@@ -126,13 +126,20 @@ def get_directory_listing_from_redis_via_lua(user, directory)
           if string.sub(name, -1) ~= "/" then
             listing[name]["Content-Type"]   = metadata["t"]
             listing[name]["Content-Length"] = tonumber(metadata["s"])
+            listing[name]["Last-Modified"]  = tonumber(metadata["m"])
           end
         end
 
         return cjson.encode(listing)
       EOF
 
-      JSON.parse(redis.eval(lua_script, nil, [user, directory]))
+      items = JSON.parse(redis.eval(lua_script, nil, [user, directory]))
+
+      items.select{|k,v| k[-1] != "/"}.each do |k,v|
```

Comment:
I would rewrite this like this to make it more readable and easier to understand it's rejecting directories (also using the Ruby convention of marking unused args with a an underscore)

```ruby
items.reject{|k, _| k.end_with? "/"}.each do |k, v|
```

Other than that LGTM!